File: app/src/main/java/com/jx/androiddemo/presenter/main/MainPresenter.java
package com.jx.androiddemo.presenter.main;

import android.content.Context;

import com.jx.androiddemo.contract.main.MainContract;
import com.jx.androiddemo.presenter.BaseRxPresenter;
import com.jx.androiddemo.tool.ToastUtil;
import com.jx.appfw.domain.interactor.main.TestAction;
import com.jx.appfw.domain.request.main.TestBean;
import com.jx.arch.domain.interactor.DefaultObserver;

import javax.inject.Inject;

public class MainPresenter extends BaseRxPresenter<MainContract.View> implements MainContract.Presenter {
    private static final String TAG = "MainPresenter";

    private Context context;

    TestAction mTestAction;


    @Inject
    MainPresenter(Context context, TestAction testAction) {
        this.context = context;
        mTestAction = testAction;
    }

    @Override
    public void doDispose() {

    }

    public void test() {
        TestBean testBean = new TestBean();
        testBean.str = "233";
        mTestAction.execute(new DefaultObserver<String>() {
            @Override
            public void onNext(String str) {
                ToastUtil.showTextToast(str);
            }
        }, testBean);
    }
}

File: app/src/main/java/com/jx/androiddemo/presenter/main/MainPresenter.kt
package com.jx.androiddemo.presenter.main

import android.content.Context
import com.jx.androiddemo.BaseApplication
import com.jx.androiddemo.constant.EventBusTag
import com.jx.androiddemo.contract.main.MainContract
import com.jx.androiddemo.event.NoticeEvent
import com.jx.androiddemo.presenter.BaseRxPresenter
import com.jx.appfw.common.DefaultObserver
import com.jx.appfw.domain.interactor.main.TestAction
import com.jx.appfw.domain.request.main.TestBean
import org.greenrobot.eventbus.EventBus

class MainPresenter(val mContext: Context) :
    BaseRxPresenter<MainContract.View>(),
    MainContract.Presenter {
    private val mTestAction: TestAction? by lazy {
        BaseApplication.getActionManager()?.getTestAction()
    }

    override fun doDispose() {
    }

    override fun test() {
        val testBean = TestBean()
        testBean.str = "test"
        mTestAction?.execute(object : DefaultObserver<String>() {
            override fun onNext(str: String) {
                getView()?.showMsg(str)
            }
        }, testBean)
    }

    override fun test2() {
        EventBus.getDefault().post(NoticeEvent(EventBusTag.TAG_TEST, arrayOf("test2")))
    }

}